backend/src/browser.py

```diff
@@ -198,6 +192,28 @@ async def _install(self, artifact: str, name: str, version: str, hash: str):
                 if res.status != 200:
                     logger.error(f"Server did not accept install count increment request. code: {res.status}")
 
+        if res_zip and version == "dev":
+            with ZipFile(res_zip) as plugin_zip:
+                plugin_json_list = [file for file in plugin_zip.namelist() if file.endswith("/plugin.json")]
+
+                if len(plugin_json_list) == 0:
+                    logger.fatal("No plugin.json found in plugin ZIP")
+                    return
+
+                elif len(plugin_json_list) > 1:
+                    logger.fatal("Multiple plugin.json found in plugin ZIP")
+                    return
+
+                else:
+                    name = json.loads(plugin_zip.read(plugin_json_list[0]))["name"]
```

Comment:
The plugin folder can differ from the plugin name in `plugin.json`. For example, `CSS Loader` installs in the `SDH-CssLoader` folder.

The plugin folder name instead should be derived from the folder name in which `plugin.json` is located within the zip.

```python
name = re.sub(r"/.+$", "", plugin_json_list[0])
```